test/NuGet.Packaging.Test/PackagesConfigReaderTests.cs

```diff
@@ -167,19 +167,108 @@ public void PackagesConfigReader_InvalidAllowedVersions()
         [Fact]
         public void PackagesConfigReader_DuplicateEntries()
         {
+            // Arrange
             var doc = XDocument.Parse(
-@"<?xml version=""1.0"" encoding=""utf-8""?>
-<packages>
-  <package id=""test1"" version=""1.0""/>
-  <package id=""test1"" version=""1.0""/>
+                @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <packages>
+                  <package id=""test1"" version=""1.0""/>
+                  <package id=""test1"" version=""1.0""/>
+
+                  <package id=""test2"" version=""1.0""/>
+                  <package id=""test2"" version=""1.0""/>
+                </packages>");
+            var reader = new PackagesConfigReader(doc);
 
-  <package id=""test2"" version=""1.0""/>
-  <package id=""test2"" version=""1.0""/>
-</packages>");
+            // Act
+            var ex = Assert.Throws<PackagesConfigReaderException>(() => reader.GetPackages());
+
+            // Assert
+            Assert.Equal(ex.Message, "There are duplicate packages: test1, test2");
+        }
+
+        [Fact]
+        public void PackagesConfigReader_DuplicateEntries_Casing()
+        {
+            // Arrange
+            var doc = XDocument.Parse(
+                @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <packages>
+                  <package id=""test1"" version=""1.0""/>
+                  <package id=""TEST1"" version=""1.0""/>
+
+                  <package id=""test2"" version=""1.0""/>
+                  <package id=""TEST2"" version=""1.0""/>
+                </packages>");
             var reader = new PackagesConfigReader(doc);
 
+            // Act
             var ex = Assert.Throws<PackagesConfigReaderException>(() => reader.GetPackages());
+
+            // Assert
             Assert.Equal(ex.Message, "There are duplicate packages: test1, test2");
         }
+
+        [Fact]
+        public void PackagesConfigReader_AllowDuplicateEntriesFailOnDuplicateVersions()
+        {
+            // Arrange
+            var doc = XDocument.Parse(
+                @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <packages>
+                  <package id=""test1"" version=""1.0""/>
+                  <package id=""test1"" version=""1.0""/>
+                  <package id=""test2"" version=""1.0""/>
+                  <package id=""test2"" version=""2.0""/>
+                </packages>");
+            var reader = new PackagesConfigReader(doc);
+
+            // Act
+            var ex = Assert.Throws<PackagesConfigReaderException>(() => 
+                reader.GetPackages(allowDuplicatePackageIds: true));
+
+            // Assert
+            Assert.Equal(ex.Message, "There are duplicate packages: test1.1.0.0");
```

Comment:
I understand the duplicate is test1, version 1.0.0 as the normalized version. But in the packages config itself it says 1.0.

Instead of using packageIdentity.ToString(), can you output something that matches what's shown on the packages.config file? So when user goes to packages.config to find the duplicate, it's easy for them to get what has been duplicated.